bindings/ergo-lib-wasm/src/transaction/reduced.rs

```diff
@@ -64,10 +77,86 @@ impl ReducedTransaction {
             .map_err(to_js)
     }
 
-    /// Returns the unsigned transation
+    /// Returns the unsigned transaction
     pub fn unsigned_tx(&self) -> UnsignedTransaction {
         self.0.unsigned_tx.clone().into()
     }
+
+    /// Generate commitment for index input with input address
+    pub fn generate_commitment_for_input(
```

Comment:
Why do we need this? I don't see it's used anywhere.

Reply:
This function returns commitments(Secret & Public) that are generated for specific input boxes in the Wasm layer.
I used it for generating commitments for inputs. Should I change it to a function that generates commitments for a whole transaction?

Reply:
This method is in Wasm layer and I don't see it's used anywhere. Why does it take a secret key as a parameter?

Reply:
And yes, we're missing commitment generation for non-reduced transactions.

Reply:
As a side note, Wasm layer is intended to be a thin and simple "conversion" layer for JS bindings. Don't forget about Rust(+ C, Java, Swift, bindings) users of this library. The core features should be implemented in ergo-lib crate so that other bindings could use it as well.

Reply:
That was a good point! I should be considered this point, I am tried in new commits considering this point and refactor the codes. I think now it's done and ready to review.